sql-plugin/src/main/scala/com/nvidia/spark/rapids/decimalExpressions.scala

```diff
@@ -16,10 +16,95 @@
 package com.nvidia.spark.rapids
 
 import ai.rapids.cudf.{ColumnVector, DType, Scalar}
+import scala.math.{max, min}
 
-import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.{CheckOverflow, Expression, PromotePrecision}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.rapids._
 import org.apache.spark.sql.types.{DataType, DecimalType, LongType}
 
+/**
+ * A GPU substitution of CheckOverflow, serves as a placeholder.
+ */
+case class GpuCheckOverflow(child: Expression) extends GpuUnaryExpression {
+  override protected def doColumnar(input: GpuColumnVector): ColumnVector =
+    input.getBase.incRefCount()
+  override def dataType: DataType = child.dataType
+}
+
+/**
+ * A GPU substitution of PromotePrecision, serves as a placeholder.
+ */
+case class GpuPromotePrecision(child: Expression) extends GpuUnaryExpression {
+  override protected def doColumnar(input: GpuColumnVector): ColumnVector =
+    input.getBase.incRefCount()
+  override def dataType: DataType = child.dataType
+}
+
+/** Meta-data for checkOverflow */
+class CheckOverflowExprMeta(
+    expr: CheckOverflow,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends UnaryExprMeta[CheckOverflow](expr, conf, parent, rule) {
+  override def convertToGpu(child: Expression): GpuExpression =
+    GpuCheckOverflow(child)
+}
+
+/** Meta-data for promotePrecision */
+class PromotePrecisionExprMeta(
+    expr: PromotePrecision,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+  extends UnaryExprMeta[PromotePrecision](expr, conf, parent, rule) {
+  override def convertToGpu(child: Expression): GpuExpression =
+    GpuPromotePrecision(child)
+}
+
+object DecimalExpressions {
```

Comment:
This entire object is never used.  Why do we need it?

Reply:
I don't think it's needed anymore, all references were stripped out with the simplification of the `PromotePrecision` operation.